plutus-contract/src/Plutus/Contract/Test/ContractModel/Interface.hs

```diff
@@ -461,7 +465,12 @@ transfer w1 w2 = QCCM.transfer (CMI.walletAddress w1) (CMI.walletAddress w2)
 -- | Register the real token corresponding to a symbolic token created
 -- in `createToken`.
 registerToken :: String -> AssetClass -> SpecificationEmulatorTrace ()
-registerToken s ac = tell [(s, ac)]
+registerToken s ac = tell @[(String, Either TxOut AssetClass)] [(s, Right ac)]
+
+-- | Register the real TxOut corresponding to a symbolic TxOut created
+-- in `createTxOut`.
+registerTxOut :: String -> CardanoAPI.TxOut CardanoAPI.CtxUTxO QCCM.Era -> SpecificationEmulatorTrace ()
```

Comment:
I would love to see this used in an example, but that doesn't have to happen in this PR.

Reply:
Yes, me too... But there were no good examples in `plutus-apps` for this, so my intention is to use the whole machinery in the `quickcheck-contractmodel-cooked` repo instead as there we have an example where adding this should be easy.